pydra/engine/specs.py

```diff
@@ -90,23 +101,66 @@ def hash(self):
         """Compute a basic hash for any given set of fields."""
         from .helpers import hash_value, hash_function
 
+        # if inp_hash already calculated and nothing has changed, the old value can be used
+        if self.changed is False and self.inp_hash:
+            # if there are files in self.files_hash, checking if they didn't changed
+            check_files = [
+                Path(val[0]).stat().st_mtime == val[1]
+                for val in self.files_hash.values()
+            ]
+            if all(check_files):
+                return self.inp_hash
+        if self.files_hash is None:
+            self.files_hash = {}
```

Comment:
shouldn't this be in `init`?

Reply:
so, should i create an init method? it could be